tests/rptest/tests/enterprise_features_license_test.py

```diff
@@ -221,6 +254,9 @@ def has_role(r: str):
         except ValueError:
             assert False, f"Unexpected status in response: {status_str}"
 
-        violation = rsp.get('violation')
-        assert violation == (not with_license), \
-            f"Expected{' no' if with_license else ''} enterprise license violation, got violation='{violation}'"
+        # TODO(oren): with this change, we can't enter a violated state at all
+        #   Probably want a test case where we bootstrap the cluster w/ all this stuff on to demonstrate the original functionality
+        #   Either that or we push a license then revoke it or something, but I don't know whether we can do that from outside
```

Comment:
We could either build in a revoke license endpoint (primarily for testing), or maybe we can store in CI/1password an expired license (generate a license with a 0day period).

Reply:
Ahh nevermind, we don't allow loading an expired license.  (Smart!)

Reply:
I guess the best you can do is _start_ in that state by disabling the built-in eval license.  Since it's generated on every boot, if you needed to test going in and out, you can start with it enabled, and then restart with it disabled.

Reply:
Yeah i don't have a great answer for this. I don't think i will until i actually start poking at the test harness